WordPress/Classes/ViewRelated/Post/Revisions/Browser/RevisionDiffsPageManager.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+
+
+protocol RevisionDiffsPageManagerDelegate: class {
+    func currentIndex() -> Int
+    func pageWillScroll(to direction: UIPageViewController.NavigationDirection)
+    func pageDidFinishAnimating(completed: Bool)
+}
+
+
+class RevisionDiffsPageManager: NSObject {
+    var viewControllers: [RevisionDiffViewController] = []
+
+    private unowned var delegate: RevisionDiffsPageManagerDelegate
+
+
+    init(delegate: RevisionDiffsPageManagerDelegate) {
+        self.delegate = delegate
+    }
+
+
+    private func index(of viewController: UIViewController?) -> Int? {
+        return viewControllers.lazy.firstIndex { $0 === viewController }
+    }
+}
+
+
+extension RevisionDiffsPageManager: UIPageViewControllerDataSource {
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        return getViewController(at: index(of: viewController),
+                                 direction: .reverse,
+                                 operation: -)
+    }
+
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerAfter viewController: UIViewController) -> UIViewController? {
+        return getViewController(at: index(of: viewController),
+                                 direction: .forward,
+                                 operation: +)
```

Comment:
Passing in the operation here is a clever bit of `DRY` however I feel it makes this harder to understand, because both `direction` and `operation` are saying the same thing, and the calling code should know as little of the implementation as necessary. Let's ditch `operation` and make `getViewController` slightly longer.